nessie/src/main/java/org/apache/iceberg/nessie/NessieUtil.java

```diff
@@ -102,6 +102,38 @@ private static String commitAuthor(Map<String, String> catalogOptions) {
         .orElseGet(() -> System.getProperty("user.name"));
   }
 
+  private static void checkAndUpdateGCProperties(
+      TableMetadata tableMetadata, Map<String, String> updatedProperties, String identifier) {
+    if (tableMetadata.propertyAsBoolean(
+        NessieTableOperations.NESSIE_GC_NO_WARNING_PROPERTY, false)) {
+      return;
+    }
+
+    // To prevent accidental deletion of files that are still referenced by other branches/tags,
+    // setting GC_ENABLED to 'false' is recommended, so that all Iceberg's gc operations like
+    // expire_snapshots, remove_orphan_files, drop_table with purge will fail with an error.
+    // `nessie-gc` CLI provides a reference-aware GC functionality for the expired/unreferenced
+    // files.
+    // Advanced users may still want to use the simpler Iceberg GC tools iff their Nessie Server
+    // contains only one branch (in which case the full Nessie history will be reflected in the
+    // Iceberg sequence of snapshots).
+    if (tableMetadata.propertyAsBoolean(
+            TableProperties.GC_ENABLED, TableProperties.GC_ENABLED_DEFAULT)
+        || tableMetadata.propertyAsBoolean(
+            TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED,
+            TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED_DEFAULT)) {
```

Comment:
I think nessie might want to use `false` explicitly here rather than rely on whatever the Iceberg default is? That's at least what the previous code was doing

Reply:
I think we need to use Iceberg defaults. The idea is to warn in case Iceberg might trigger the behaviour controlled by those properties, hence we need to apply the same default.

Reply:
it's up to you, I just wanted to point that out :)

Reply:
Should we add a table-default for this one as well?

Reply:
For ref: https://github.com/apache/iceberg/blob/81bf8d30766b1b129b87abde15239645cb127046/core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java#L417-L420

Reply:
> Should we add a table-default for this one as well?

The Iceberg default is `false` for this one. If the default changes `testTableMetadataFilesCleanupDisable()` will catch that... but I'll add a Nessie-specific default just in case.

Reply:
Added table default for `METADATA_DELETE_AFTER_COMMIT_ENABLED` in NessieCatalog